benchmarks/src/csv/load.rs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::path::PathBuf;
+
+use crate::util::{BenchmarkRun, CommonOpt};
+use datafusion::{
+    common::Result,
+    prelude::{CsvReadOptions, SessionContext},
+};
+
+use datafusion_common::instant::Instant;
+use structopt::StructOpt;
+
+use super::data::DataOpt;
+
+#[derive(Debug, StructOpt, Clone)]
+#[structopt(verbatim_doc_comment)]
+pub struct RunOpt {
+    /// Common options
+    #[structopt(flatten)]
+    common: CommonOpt,
+
+    /// Create data files
+    #[structopt(flatten)]
+    data: DataOpt,
+
+    /// Path to machine readable output file
+    #[structopt(parse(from_os_str), short = "o", long = "output")]
+    output_path: Option<PathBuf>,
+}
+
+impl RunOpt {
+    pub async fn run(self) -> Result<()> {
+        let test_file = self.data.build()?;
+        let mut rundata = BenchmarkRun::new();
```

Comment:
One thing I would like to request is that we split the data generation from the query. 

Given this setup, rerunning the benchmarks will likely be dominated by the time it takes to regenerate the input which will be quite slow